clients/src/main/java/org/apache/kafka/clients/producer/internals/DefaultPartitioner.java

```diff
@@ -33,16 +29,17 @@
  * <ul>
  * <li>If a partition is specified in the record, use it
  * <li>If no partition is specified but a key is present choose a partition based on a hash of the key
- * <li>If no partition or key is present choose a partition in a round-robin fashion
+ * <li>If no partition or key is present choose the sticky partition that changes when the batch is full.
```

Comment:
It would be nice to add a comment here like "see KIP-480 for details" so that people would know where to look to find more information.